app/events/domains/proxy_domains_changed_event.rb

```diff
@@ -0,0 +1,29 @@
+require 'uri'
+
+class Domains::ProxyDomainsChangedEvent < BaseEventStoreEvent
+  def self.create(proxy)
+    new(
+      proxy:    proxy,
+      staging_domains: extract_domain(proxy.sandbox_endpoint),
+      production_domains: extract_domain(proxy.endpoint),
+
+      metadata: {
+        provider_id: (provider_id = proxy.provider&.id),
```

Comment:
This loads the service and the provider:
https://github.com/3scale/porta/blob/49032f8f4e7c80bbb6a37bf0858190ffa0b45eae/app/models/proxy.rb#L471-L473
So if all you want is the id, imo you should just do `(provider_id = proxy.tenant_id || proxy.provider&.id)` so if `tenant_id` is there, it will be more efficient 😄

Reply:
I'd not find it very clear assigning a tenant_id into a provider_id. Sure it is more efficient, but also makes less sense :)